### PR TITLE
Modernise forms

### PR DESCRIPTION
**Pull request by janfaracik**
This PR updates the UI of this plugin to match the latest UI and language changes in Jenkins.

**What's changed**
* Updated some forms classes to fit in with the updated Jenkins UI
* Updated some language to be clearer and consistent with Jenkins
* Delete buttons now use Jenkins' built in component for consistency
* Removed a superfluous delete button

**Screenshots**

Screenshot 2021-12-18 at 23 38 59

_to_

image

---

Screenshot 2021-12-18 at 23 39 21

_to_

image

---

* CodeMirror textboxes don't have a border at the moment on current Jenkins weekly builds, this will be fixed soon without needing to change the plugin

### CODE REVIEW

**Review by timja — COMMENTED** [review]
Looks like this PR isn't compatible with the styling without the modernise forms PR, would be nice if it was so that it was forward compatible but the PR can probably wait till it's released too

2.303.1

<details>
<summary>Before</summary>

image

</details>

<details>
<summary>After</summary>

image


</details>

File: src/main/resources/org/jenkinsci/plugins/configfiles/maven/MavenSettingsConfig/edit-config.jelly [hunk 1]
```diff
@@ -26,11 +26,6 @@ THE SOFTWARE.
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define"
 	xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" >
-<style type="text/css">
-div.credentialssection {
-  border-bottom: 1px solid black;
-}
-</style>
```
Comment (daniel-beck) [inline src/main/resources/org/jenkinsci/plugins/configfiles/maven/MavenSettingsConfig/edit-config.jelly:33]: 🎉  I was very surprised when I saw these random divider lines on the UI 👍

File: src/main/resources/org/jenkinsci/plugins/configfiles/maven/GlobalMavenSettingsConfig/edit-config.jelly [hunk 2]
```diff
@@ -26,15 +26,7 @@ THE SOFTWARE.
...
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define"
 	xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
-<style type="text/css">
-div.credentialssection {
-  border-bottom: 1px solid black;
-}
-</style>
 			<j:set var="instance" value="${config}"/>
-	   		<p>
-	   		 ${%description}
```
Comment (daniel-beck) [inline src/main/resources/org/jenkinsci/plugins/configfiles/maven/GlobalMavenSettingsConfig/edit-config.jelly:36]: FTR there is no localized text for this, so it makes sense to remove without replacement:
Screenshot 2022-01-08 at 23 08 15

**Comment by alecharp** [comment]
@janfaracik thank you for the PR however, because there are conflicts, I cannot review as is. Also, please address the comment from @timja.

**Comment by janfaracik** [comment]
> Looks like this PR isn't compatible with the styling without the modernise forms PR, would be nice if it was so that it was forward compatible but the PR can probably wait till it's released too
> 
> 2.303.1
> 
> Before
> image
> 
> After
> image

I've updated it with a rather messy if condition to add some fallback styling for Jenkins releases that don't have the new forms styling.

image

```
<!-- Remove when Jenkins LTS release is no longer supported by update center-->
<j:if test="${h.version.startsWith('2.289.') or h.version.startsWith('2.303.') or h.version.startsWith('2.319.')}">
```

It works but it isn't really a good way of going about it - I'm all ears for better ways to do the check

Thanks :)

File: src/main/resources/org/jenkinsci/plugins/configfiles/ConfigFilesUI/selectprovider.jelly [hunk 3]
```diff
@@ -28,6 +28,20 @@ THE SOFTWARE.
 	<l:layout title="${%filetype}" norefresh="true">
 		<cf:sitepanel />
 
+		<!-- Remove when Jenkins LTS release is no longer supported by update center-->
+		<j:if test="${h.version.startsWith('2.289.') or h.version.startsWith('2.303.') or h.version.startsWith('2.319.')}">
```
Comment (timja) [inline src/main/resources/org/jenkinsci/plugins/configfiles/ConfigFilesUI/selectprovider.jelly:32]: can probably make this always here with a TODO comment?

it's on it's own standalone page

Comment (timja) [reply src/main/resources/org/jenkinsci/plugins/configfiles/ConfigFilesUI/selectprovider.jelly:32]: I meant remove the if statement not change to TODO

Comment (janfaracik) [reply src/main/resources/org/jenkinsci/plugins/configfiles/ConfigFilesUI/selectprovider.jelly]: Ah I misunderstood - hope the latest commit is more inline with what you're thinking.

**Review by NotMyFault — APPROVED** [review]
Looks great, same on the - now bumped - plugin baseline 👍🏻

**Pull request merged**